core/ledger/kvledger/snapshot.go

```diff
@@ -371,3 +384,28 @@ func verifyFileHash(dir, file string, expectedHashInHex string, hashProvider led
 	}
 	return nil
 }
+
+type mostRecentCollectionConfigFetcher struct {
```

Comment:
maybe a ledger level integration test is needed as there are many components involved. Can be done in a separate PR as well.

Reply:
Not sure that I understand this comment...
There is a unit test (`TestMostRecentCollectionConfigFetcher`) for this code in this package. One high level test (`TestSnapshotGenerationAndNewLedgerCreation`) that exercises this in a e2e flow, also exists in this package. There will be final integration tests related to sanpshotting.
What else is the suggestion?

Reply:
The prugeMgr is not tested in the high-level test. What I suggested what that we need a test at `kvledger/test`s to ensure that `pvtdatapolicy`, `privacyenabledstate`, `purgemgr`, and other dependencies play well together while constructing the purgemgr. For example, the BTLPolicy used in the prugeMgr is not the real one and it has different behavior.

Reply:
OK, got puzzled why you singled out this particular piece of code. There are many others aspects and scenarios that are unit tested and will get tested in the integration tests (at kvledger/tests and real integration tests). You saw one bug that was discovered with the real channel config. Yes, we are getting closer.